torch/nn/modules/sparse.py

```diff
@@ -119,6 +124,14 @@ def __repr__(self):
         s += ')'
         return s.format(name=self.__class__.__name__, **self.__dict__)
 
+    @classmethod
+    def load_pretrained(cls, embeddings, freeze=True):
+        rows, cols = embeddings.shape
```

Comment:
Check that `embeddings.dim() == 2` and raise a nice error message saying that it was expected ot be 2D